x-pack/plugins/uptime/server/lib/alerts/status_check.ts

```diff
@@ -321,19 +333,24 @@ export const statusCheckAlertFactory: UptimeAlertTypeFactory = (_server, libs) =
       const mergedIdsByLoc = getUniqueIdsByLoc(downMonitorsByLocation, availabilityResults);
 
       mergedIdsByLoc.forEach((monIdByLoc) => {
-        const alertInstance = alertInstanceFactory(MONITOR_STATUS.id + monIdByLoc);
-
         const availMonInfo = availabilityResults.find(
-          ({ monitorId, location }) => monitorId + location === monIdByLoc
+          ({ monitorId, location }) => getMonIdByLoc(monitorId, location) === monIdByLoc
         );
 
         const downMonInfo = downMonitorsByLocation.find(
-          ({ monitorId, location }) => monitorId + location === monIdByLoc
+          ({ monitorId, location }) => getMonIdByLoc(monitorId, location) === monIdByLoc
         )?.monitorInfo;
 
-        const monitorSummary = getMonitorSummary(downMonInfo || availMonInfo?.monitorInfo!);
+        const monitorInfo = downMonInfo || availMonInfo?.monitorInfo!;
+
+        const monitorSummary = getMonitorSummary(monitorInfo);
         const statusMessage = getStatusMessage(downMonInfo!, availMonInfo!, availability);
 
+        // replace url special characters with -
+        const urlText = monitorInfo.url?.full?.replace(/[^A-Z0-9]+/gi, '_').toLowerCase();
+
+        const alertInstance = alertInstanceFactory(urlText + '-' + monIdByLoc);
```

Comment:
A better formula IMHO would be `${monitor.name}-${url}-${monitor.id}` where we prefer showing the name since that's the best human readable thing if it's around. If there's no name, obviously we don't show it.

With synthetics URLs will be very non-unique, since many journeys will start at a homepage, for instance, but accomplish different tasks (sign up, check out, etc.).

Reply:
done in [`eb400a9` (#81736)](https://github.com/elastic/kibana/pull/81736/commits/eb400a9d1dfad99ff521dd9847b5b63ddeeaa48d)